packages/firestore/lite/test/integration.test.ts

```diff
@@ -1045,4 +1076,111 @@ describe('withConverter() support', () => {
       expect(refEqual(docRef, docRef2)).to.be.false;
     });
   });
+
+  it('requires the correct converter for Partial usage', async () => {
+    return withTestDb(async db => {
+      const coll = collection(db, 'posts');
+      const ref = doc(coll, 'post').withConverter(postConverter);
+      const batch = writeBatch(db);
+      expect(() =>
+        batch.set(ref, { title: 'olive' }, { merge: true })
+      ).to.throw(
+        'Function WriteBatch.set() called with invalid data ' +
+          '(via `toFirestore()`). Unsupported field value: undefined ' +
+          '(found in field author in document posts/post)'
+      );
+    });
+  });
+
+  it('WriteBatch.set() supports partials with merge', async () => {
```

Comment:
Can you move these tests to https://github.com/firebase/firebase-js-sdk/blob/master/packages/firestore/lite/test/integration.test.ts#L253, which allows you to write one test for WriteBatch, Transaction and set().

Reply:
done.